python/nav/util.py

```diff
@@ -509,3 +509,20 @@ def _range_to_str(x, y):
             return str(x)
         else:
             return "{}-{}".format(x, y)
+
+
+def check_log_level(logger, loglevel):
+    """
+    Check that the level of the logger is equal to or below loglevel
+
+    Use before possibly expensive logging operations, like calling logging in
+    a loop or building an expensive object that will be logged and thrown away.
+
+    Avoid this classic mistake:
+
+        logger.info("%s", expensive_function())
+    """
```

Comment:
Another pattern for this is to change `expensive_function()` to be some object that runs the expensive stuff only when something attempts to convert it to a string - as the string interpolation of the logger output is only run if the log level matches.

Reply:
True, but that's more magic and thus prone to disappearing when someone feels like cleaning out seemingly unneccessary code.